kafka_utils/kafka_rolling_restart/version_precheck.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+from functools import partial
+
+from fabric.api import execute
+from fabric.api import hide
+from fabric.api import settings
+from fabric.api import sudo
+from fabric.api import task
+
+from .task import PreStopTask
+from .task import TaskFailedException
+
+CHECK_KAFKA_PACKAGE = "dpkg -s {} | grep '^Version: {}'"
+KAFKA_UPDATE_CMD = "run-puppet"
+
+
+class VersionPrecheck(PreStopTask):
```

Comment:
Same as above. We should probably also rename this to CheckVersionOrUpgrade because VersionPrecheck doesn't imply that a change (upgrade) is performed when the version is not matching.